easybuild/tools/module_naming_scheme/hierarchical_mns.py

```diff
@@ -47,10 +47,10 @@
 
 # note: names in keys are ordered alphabetically
 COMP_NAME_VERSION_TEMPLATES = {
-    'icc,ifort': ('intel', '%(icc)s'),
-    'Clang,GCC': ('Clang-GCC', '%(Clang)s-%(GCC)s'),
-    'CUDA,GCC': ('GCC-CUDA', '%(GCC)s-%(CUDA)s'),
-    'xlc,xlf': ('xlcxlf', '%(xlc)s'),
+    'icc,ifort': ('intel', ('icc',)),
+    'Clang,GCC': ('Clang-GCC', ('Clang', 'GCC')),
+    'CUDA,GCC': ('GCC-CUDA', ('GCC', 'CUDA')),
+    'xlc,xlf': ('xlcxlf', ('xlc',)),
```

Comment:
@thiell I think we should try and avoid changing the format here, since other module naming schemes may be relying on this...

The only reason you did this is to have access to the names used to resolve the templates with, and make sure they're all available before resolving the templates, right?

In that case, why not use the names in the `key.split(',')` instead? Then there's no need to change the format here?